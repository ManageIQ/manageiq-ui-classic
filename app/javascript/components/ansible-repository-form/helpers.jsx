const GitUrlParse = require('git-url-parse');

export const urlValidation = (values) => {
  const errors = {};
  var error = false;
  try {
    const parsed = values.scm_url ? GitUrlParse(values.scm_url) : {};
    if ((parsed.protocol !== 'http'
          && parsed.protocol !== 'https'
          && parsed.protocol !== 'file'
          && parsed.protocol !== 'ssh')
          || parsed.pathname === '') {
      error = true;
    }
    if (parsed.protocols.length > 2 || (parsed.protocols.length === 1 &&  parsed.protocols[0] !== parsed.protocol)) {
      error = true;
    }
  } catch (err) {
    error = true;
  }
  if (error) {
    errors.scm_url = __('URL must include a protocol (http://, https:// or file://) with path or be a valid SSH path (user@server:path or ssh://user@address:port/path)');
  }
  return errors;
};
