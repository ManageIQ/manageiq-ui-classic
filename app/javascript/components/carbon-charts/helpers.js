export const sampleData = [{
  group: 'Dataset 1',
  key: '1',
  value: 100,
},
{
  group: 'Dataset 1',
  key: '2',
  value: 400,
},
{
  group: 'Dataset 1',
  key: '4',
  value: 500,
},
{
  group: 'Dataset 1',
  key: '3',
  value: 200,
},
{
  group: 'Dataset 1',
  key: '5',
  value: 150,
},
{
  group: 'Dataset 2',
  key: '1',
  value: 280,
},
{
  group: 'Dataset 2',
  key: '2',
  value: 350,
},
{
  group: 'Dataset 2',
  key: '3',
  value: 700,
},
{
  group: 'Dataset 2',
  key: '5',
  value: 600,
},
{
  group: 'Dataset 2',
  key: '5',
  value: 200,
},
{
  group: 'Dataset 3',
  key: '4',
  value: 100,
},
{
  group: 'Dataset 3',
  key: '2',
  value: 500,
},
{
  group: 'Dataset 3',
  key: '1',
  value: 200,
}];

export const pieData = [{
  group: 'Dataset 1',
  key: '1',
  value: 100,
},
{
  group: 'Dataset 2',
  key: '1',
  value: 280,
},
{
  group: 'Dataset 3',
  key: '4',
  value: 100,
},
{
  group: 'Dataset 4',
  key: '2',
  value: 500,
}];

// convert report data to carbon chart data format.
export const getConvertedData = (data) => {
  if (data && data.data.columns && data.data && data.miq && data.miq.category_table && data.miq.name_table) {
    const columnsData = data.data.columns;
    const dataGroups = data.miq.category_table;
    const rowsData = data.miq.name_table;
    const arr = [];
    if (columnsData.length > 0) {
      columnsData.forEach((items) => {
        items.forEach((item, i) => {
          const obj = {};
          if (i !== 0 && rowsData[items[0]]) {
            obj.group = rowsData[items[0]];
            obj.key = dataGroups[i - 1];
            obj.value = item;
            arr.push(obj);
          }
        });
      });
    }
    return arr;
  }
  return [];
};

export const getLineConvertedData = (data) => {
  const columnsData = data.data.columns;
  const dataGroups = data.miq.category_table;
  const rowsData = data.miq.name_table;
  const arr = [];
  columnsData.forEach((items) => {
    items.forEach((item, i) => {
      const obj = {};
      if (i !== 0 && rowsData[items[0]]) {
        obj.group = rowsData[items[0]];
        obj.key = new Date(dataGroups[i - 1]);
        obj.value = item;
        arr.push(obj);
      }
    });
  });
  return arr;
};
