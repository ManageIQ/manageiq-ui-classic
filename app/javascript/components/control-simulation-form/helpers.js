// Helper Functions

var showPassedPolicies = true;
var showFailedPolicies = true;
var showOutOfScopeItems = true;

const convertApi = (apiData, showPassedPolicies, showFailedPolicies, showOutOfScopeSelected) => {
  showPassedPolicies = showPassedPolicies;
  showFailedPolicies = showFailedPolicies;
  showOutOfScopeItems = showOutOfScopeSelected;

  if (apiData === undefined) {
    return '';
  }
  console.log('api Data');
  console.log(apiData);

  const convToString = [{
    key: 'root',
    text: apiData.name,
    icon: 'fa fa-star',
    class: 'no-cursor',
    selectable: false,
    state: {
      expanded: true,
    },
    nodes: findAllVms(apiData.task_results),
  }];
  return convToString;
};

const findAllVms = (allVms) => {
  // TODO: test this, there are cases with more than one vm (By Clusters -> Cluster)
  const holdAllVms = [];

  allVms.forEach((vmData) => {
    const vmObj = {
      key: `xx-${vmData.id}`,
      text: `\u003cstrong\u003eVM:\u003c/strong\u003e ${vmData.name}`,
      icon: 'pficon pficon-virtual-machine',
      class: 'no-cursor',
      selectable: false,
      nodes: findAllProfiles(vmData.profiles, `xx-${vmData.id}`),
      state: {
        expanded: false,
      },
    };
    holdAllVms.push(vmObj);
  });
  return holdAllVms;
};

const findAllProfiles = (allProfiles, upperKey) => {
  const holdAllPolicies = [];

  allProfiles.forEach((profileData) => {
    const profileObj = {
      key: `${upperKey}_xx-${profileData.id}`,
      text: `\u003cstrong\u003eProfile:\u003c/strong\u003e ${profileData.name}`,
      icon: determineIcon(profileData.result),
      class: 'no-cursor',
      selectable: false,
      nodes: findAllPolicies(profileData.policies, `${upperKey}_xx-${profileData.id}`),
      state: {
        expanded: false,
      },
    };
    holdAllPolicies.push(profileObj);
  });
  return holdAllPolicies;
};

const findAllPolicies = (allPolicies, upperKey) => {
  const holdAllPolicies = [];
  allPolicies.forEach((policyData) => {
    // NOTE: you cannot give the tree a {} option it will break
    const nodes = [];
    if (policyData.scope) {
      nodes.push(findAllScope(policyData.scope, `${upperKey}_xx-${policyData.id}`));
    }
    if (policyData.conditions.length !== 0) {
      nodes.push(findAllConditions(policyData.conditions, `${upperKey}_xx-${policyData.id}`));
    }
    if (policyData.actions.length !== 0) {
      nodes.push(findAllActions(policyData.actions, `${upperKey}_xx-${policyData.id}`));
    }
    const policyObj = {
      key: `${upperKey}_xx-${policyData.id}`,
      text: `\u003cstrong\u003ePolicy:\u003c/strong\u003e ${policyData.description}`,
      icon: determineIcon(policyData.result),
      class: 'no-cursor',
      selectable: false,
      nodes,
      state: {
        expanded: false,
      },
    };

    // NOTE: policyData.result can be: deny || allow || N/A
    if (policyData.result === 'deny' && showFailedPolicies === true) {
      holdAllPolicies.push(policyObj);
    } else if (policyData.result === 'allow' && showPassedPolicies === true) {
      holdAllPolicies.push(policyObj);
    } else if (policyData.result === 'N/A' && showOutOfScopeItems === true) {
      holdAllPolicies.push(policyObj);
    }
  });
  return holdAllPolicies;
};

const findAllConditions = (allConditions, upperKey) => {
  let holdAllConditions = {};

  allConditions.forEach((conditionData) => {
    const conditionObj = {
      key: `${upperKey}_xx-${conditionData.id}`,
      text: `\u003cstrong\u003eCondition:\u003c/strong\u003e ${conditionData.description}`,
      icon: determineIcon(conditionData.result),
      class: 'no-cursor',
      selectable: false,
      nodes: [findAllExpressions(conditionData.expression, `${upperKey}_xx-${conditionData.id}`)],
      state: {
        expanded: false,
      },
    };
    holdAllConditions = conditionObj;
  });
  return holdAllConditions;
};

const findAllActions = (allActions, upperKey) => {
  let holdAllActions = {};

  allActions.forEach((actionData) => {
    const actionObj = {
      key: `${upperKey}_xx-${actionData.id}`,
      text: `\u003cstrong\u003eAction:\u003c/strong\u003e ${actionData.description}`,
      icon: determineIcon(actionData.result),
      class: 'no-cursor',
      selectable: false,
      state: {
        expanded: false,
      },
    };
    holdAllActions = actionObj;
  });
  return holdAllActions;
};

const findAllExpressions = (allExpressions, upperKey) => {
  const dynamicallyBuildText = createSentance(allExpressions, '');
  const holdExpression = {
    key: `${upperKey}_xx-`, // NOTE: this doesnt have an id
    text: `\u003cstrong\u003eExpression:\u003c/strong\u003e ${determineTextColor(allExpressions.result, dynamicallyBuildText)}`,
    tooltip: dynamicallyBuildText,
    icon: determineIcon(allExpressions.result),
    class: 'no-cursor',
    selectable: false,
    state: {
      expanded: false,
    },
  };
  return holdExpression;
};

const findAllScope = (allScope, upperKey) => {
  const dynamicallyBuildText = createSentance(allScope, '');
  const holdScope = {
    key: `${upperKey}_xx-`, // NOTE: this doesnt have an id
    text: `\u003cstrong\u003eScope:\u003c/strong\u003e ${determineTextColor(allScope.result, dynamicallyBuildText)}`,
    tooltip: dynamicallyBuildText,
    icon: determineIcon(allScope.result), // TODO: Is scope always red x ?
    class: 'no-cursor',
    selectable: false,
    state: {
      expanded: false,
    },
  };
  return holdScope;
};

export { convertApi };

// ==================================================================== //

const createSentance = (object, runOnSentence) => {
  // TODO: Test more
  let result = runOnSentence;
  // var useAnd = result == "" ? "" : " && ";
  for (const key in object) {
    if (key == 'result') {
      return result;
    } if (object[key].field != undefined) {
      result = `${result + (result === '' ? '' : ' and ') + object[key].field} ${key} ${object[key].value}`;
    } else {
      result = result + (result === '' ? '' : ' and ') + createSentance(object[key], runOnSentence);
    }
  }
  return result;
};

const determineIcon = (resultValue) => {
  // "icon" :
  // "fa fa-ban"                    <- grey crossed circle
  // "pficon pficon-ok"             <- green check circle
  // "pficon pficon-error-circle-o" <- red x circle
  switch (resultValue) {
    case 'deny':
      return 'pficon pficon-error-circle-o';
    case 'N/A':
      return 'fa fa-ban';
    case 'allow':
      return 'pficon pficon-ok';
    case true: // this happens in the scope / expression level
      return 'pficon pficon-ok'; // "fa fa-bell";
    case false:
      return 'pficon pficon-error-circle-o'; // "fa fa-bell-o";
    default:
      return 'fa fa-ban'; // "fa fa-bell-slash";
  }
};

const determineTextColor = (resultValue, text) => {
  switch (resultValue) {
    case true: // this happens in the scope / expression level
      return `\u003cfont color="green"\u003e${text}\u003c/font\u003e`;
    case false:
      return `\u003cfont color="red"\u003e${text}\u003c/font\u003e`;
    default:
      return text;
  }
};
