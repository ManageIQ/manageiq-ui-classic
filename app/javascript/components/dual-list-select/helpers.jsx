/* *********** HELPERS FOR FORMS *********** */

/* Get keys of options array
  Options:
    [
      {key: 'key', label: 'label'},
      {key: 'key1', label: 'label1'},
    ]
  returns ['key', 'key1']
*/
export const getKeys = value => (Array.isArray(value) ? value.map(({ key }) => key) : []);

/* Returns array of options excluding value options */
export const filterOptions = (options, value) => options.filter(({ key }) => !value.includes(key));

/* Returns items from newValue which are not in value */
export const filterValues = (newValue, value) => newValue.filter(item => !value.includes(item));
