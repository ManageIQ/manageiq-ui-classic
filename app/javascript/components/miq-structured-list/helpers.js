/* eslint-disable no-eval */
import { TreeViewRedux } from '../tree-view';
import CatalogResource from '../data-tables/catalog-resource';
import MiqDataTable from '../miq-data-table';

export const InputTypes = {
  TEXTAREA: 'text_area',
  CHECKBOX: 'checkbox',
  COMPONENT: 'component',
  DROPDOWN: 'dropdown',
  CODEMIRROR: 'code_mirror',
  MARKDOWN: 'markdown',
};

export const DynamicReactComponents = {
  TREE_VIEW_REDUX: TreeViewRedux,
  CATALOG_RESOURCE: CatalogResource,
  MIQ_DATA_TABLE: MiqDataTable,
};

const dataType = (data) => (data ? data.constructor.name.toString() : undefined);

export const isObject = (item) => dataType(item) === 'Object';

export const isArray = (item) => dataType(item) === 'Array';

export const isSubItem = (item) => item.sub_items && item.sub_items.length > 0;

/** When the props named mode includes 'miq_summary', then then summary contains row/cell click events. */
export const hasClickEvents = (mode) => mode.includes('miq_summary');

export const hasInput = (item) => item && Object.keys(item).includes('input');

export const rowClickEvent = (rows, index) => {
  const item = rows[index];
  if (item && item.onclick) {
    miqSparkleOn();
    eval(item.onclick);
    miqSparkleOff();
  }
};
