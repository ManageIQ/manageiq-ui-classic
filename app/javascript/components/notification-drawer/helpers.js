export const newCountText = function(count) {
  return sprintf(n__('%d New', '%d New', count), count);
};

export const getNotficationStatusIconName = notification => (
  {
    info: 'info',
    error: 'error',
    danger: 'error',
    warning: 'warning',
    success: 'success',
    ok: 'success',
  }[notification.type] || '');

export const viewDetails = function(notification) {
  window.location.href = `/restful_redirect?${$.param(notification.data.link)}`;
};
