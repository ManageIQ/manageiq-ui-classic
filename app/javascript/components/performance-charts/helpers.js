export const getYAxisValue = (format, value) => {
  // eslint-disable-next-line no-useless-escape
  if (format) {
    const tmp = /^([0-9\,\.]+)(.*)/.exec(ManageIQ.charts.formatters[format.function].c3(format.options)(value));
    return [`${numeral(tmp[1]).value()}${tmp[2]}`];
  }
  return value;
};
