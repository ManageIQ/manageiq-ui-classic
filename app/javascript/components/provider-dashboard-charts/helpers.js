import moment from 'moment';

export const getConvertedData = (data, units) => {
  if (data && data.xData && data.yData) {
    const columnsData = data.xData;
    const rowsData = data.yData;
    const arr = [];
    if (columnsData.length > 0) {
      columnsData.forEach((item, i) => {
        const obj = {};
        obj.group = units;
        obj.date = moment(item).tz(ManageIQ.timezone || 'UTC').format('MM/DD/YYYY HH:mm:ss z')
        obj.value = rowsData[i];
        arr.push(obj);
      });
    }
    return arr;
  }
  return [];
};

export const getLatestValue = (data) => {
  if (data && data.yData) {
    return data.yData[data.yData.length - 1];
  }
  return 0;
};

export const getHeatMapData = (data) => {
  if (data) {
    const arr = [];
    data.forEach((item) => {
      const obj = {};
      obj.percent = item.percent;
      obj.total = item.total;
      obj.provider = item.provider;
      obj.node = item.node;
      obj.value = item.percent * 100;
      obj.unit = item.unit;
      arr.push(obj);
    });
    return arr;
  }
  return [];
};

export const getPodsData = (data, createdlabel, deletedLabel) => {
  if (data && data.xData && data.yCreated && data.yDeleted) {
    const columnsData = data.xData;
    const createdData = data.yCreated;
    const deletedData = data.yDeleted;
    const arr = [];
    if (createdData.length > 0) {
      createdData.forEach((item, i) => {
        const obj = {};
        obj.group = createdlabel;
        obj.date = columnsData[i];
        obj.value = item;
        arr.push(obj);
      });
    }
    if (deletedData.length > 0) {
      deletedData.forEach((item, i) => {
        const obj = {};
        obj.group = deletedLabel;
        obj.date = columnsData[i];
        obj.value = item;
        arr.push(obj);
      });
    }
    return arr;
  }
  return [];
};
