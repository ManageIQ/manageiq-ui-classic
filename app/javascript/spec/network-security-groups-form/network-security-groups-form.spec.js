import React from 'react';
import toJson from 'enzyme-to-json';
import fetchMock from 'fetch-mock';
import { shallow } from 'enzyme';
import NetworkSecurityGroupsForm from '../../components/network-security-groups-form';
import '../helpers/miqAjaxButton';

require('../helpers/set_fixtures_helper.js');
require('../helpers/old_js_file_require_helper.js');
require('../helpers/miqSparkle.js');
require('../helpers/miqAjaxButton.js');

jest.mock('../../helpers/miq-redirect-back', () => jest.fn());

describe('Network Router Interfaces Form Component', () => {
  let submitSpyMiqSparkleOn;
  let submitSpyMiqSparkleOff;
  let spyMiqAjaxButton;

  const securityGroupId = '82';
  const providers = [
    {
      id: '54',
      parent_ems_id: '22',
      tenant_id: '1',
      type: 'ManageIQ::Providers::Redhat::NetworkManager',
      zone_id: '9',
      name: 'RHV Network Manager',
    },
    {
      id: '31',
      parent_ems_id: '30',
      tenant_id: '1',
      type: 'ManageIQ::Providers::Openstack::NetworkManager',
      zone_id: '8',
      name: 'OpenStack Director Network Manager',
    },
  ];
  const firewallRules = [
    {
      created_on: '2017-10-06T16:57:17Z',
      direction: 'outbound',
      display_name: null,
      ems_ref: '12ff3723-d9a3-423c-9033-6eed53d0811a',
      enabled: null,
      end_port: null,
      group: null,
      host_protocol: '',
      id: '461',
      name: null,
      network_protocol: 'IPV4',
      port: null,
      required: null,
      resource_id: '82',
      resource_type: 'SecurityGroup',
      source_ip_range: null,
      source_security_group_id: null,
      updated_on: '2017-10-06T16:57:17Z',
    },
    {
      created_on: '2017-10-06T16:57:17Z',
      direction: 'inbound',
      display_name: null,
      ems_ref: '4428d6bf-ab6d-4de9-a8eb-74c4aeed2595',
      enabled: null,
      end_port: 22,
      group: null,
      host_protocol: 'TCP',
      id: '463',
      name: null,
      network_protocol: 'IPV4',
      port: 22,
      required: null,
      resource_id: '82',
      resource_type: 'SecurityGroup',
      source_ip_range: '0.0.0.0/0',
      source_security_group_id: null,
      updated_on: '2017-10-06T16:57:17Z',
    },
  ];

  const securityGroup = {
    cloud_tenant: {
      created_at: '2017-10-06T16:56:43Z',
      description: '',
      ems_id: '7',
      ems_ref: '074ea3f5f49c47dc87d21c3f1b4d53fa',
      enabled: true,
      id: '1',
      name: 'cloud-user-demo',
      parent_id: null,
      type: 'ManageIQ::Providers::Openstack::CloudManager::CloudTenant',
      updated_at: '2017-10-06T16:56:43Z',
    },
    description: '',
    ems_id: '8',
    ext_management_system: {
      name: 'OpenStack Network Manager',
    },
    firewall_rules: firewallRules,
    href: 'http://localhost:3000/api/security_groups/82',
    id: '82',
    name: 'asimonel-sg',
  };

  beforeEach(() => {
    submitSpyMiqSparkleOn = jest.spyOn(window, 'miqSparkleOn');
    submitSpyMiqSparkleOff = jest.spyOn(window, 'miqSparkleOff');
    spyMiqAjaxButton = jest.spyOn(window, 'miqAjaxButton');
  });

  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
    submitSpyMiqSparkleOn.mockRestore();
    submitSpyMiqSparkleOff.mockRestore();
    spyMiqAjaxButton.mockRestore();
  });

  it('should render add security group form', (done) => {
    const wrapper = shallow(<NetworkSecurityGroupsForm />);
    fetchMock.getOnce(
      '/api/providers?&expand=resources&filter[]=supports_create_security_group=true',
      providers,
    );
    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });

  it('should render edit security group form', (done) => {
    const wrapper = shallow(<NetworkSecurityGroupsForm securityGroupId={securityGroupId} />);
    fetchMock.getOnce(
      `/api/security_groups/${securityGroupId}?attributes=name,ext_management_system.name,description,cloud_tenant,firewall_rules,ems_id`,
      securityGroup
    );
    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });

  it('should add security group', (done) => {
    const wrapper = shallow(<NetworkSecurityGroupsForm />);
    const addSecurityGroup = {
      ems_id: '8', name: '00', description: '1', cloud_tenant_id: '2', firewall_rules: [],
    };
    miqAjaxButton(
      `/security_group/update/${securityGroupId}?button=save`,
      addSecurityGroup,
      { complete: false }
    );
    expect(spyMiqAjaxButton).toHaveBeenCalledWith(
      `/security_group/update/${securityGroupId}?button=save`,
      addSecurityGroup,
      { complete: false }
    );
    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });

  it('should edit security group', (done) => {
    const wrapper = shallow(<NetworkSecurityGroupsForm securityGroupId={securityGroupId} />);
    const editSecurityGroup = {
      cloud_tenant: {
        name: 'cloud-user-demo',
      },
      description: '',
      ems_id: '8',
      ext_management_system: {
        name: 'OpenStack Network Manager',
      },
      firewall_rules: firewallRules,
      firewall_rules_delete: false,
      href: 'http://localhost:3000/api/security_groups/82',
      id: '82',
      name: 'asimonel-sg',
    };
    miqAjaxButton(
      `/security_group/update/${securityGroupId}?button=save`,
      editSecurityGroup,
      { complete: false }
    );
    expect(spyMiqAjaxButton).toHaveBeenCalledWith(
      `/security_group/update/${securityGroupId}?button=save`,
      editSecurityGroup,
      { complete: false }
    );
    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });

  it('should edit security group firewall rules', (done) => {
    const wrapper = shallow(<NetworkSecurityGroupsForm securityGroupId={securityGroupId} />);
    const editedFirewallRules = [
      {
        created_on: '2017-10-06T16:57:17Z',
        direction: 'outbound',
        display_name: null,
        ems_ref: '12ff3723-d9a3-423c-9033-6eed53d0811a',
        enabled: null,
        end_port: null,
        group: null,
        host_protocol: '',
        id: '461',
        name: null,
        network_protocol: 'IPV4',
        port: null,
        required: null,
        resource_id: '82',
        resource_type: 'SecurityGroup',
        source_ip_range: null,
        source_security_group_id: null,
        updated_on: '2017-10-06T16:57:17Z',
      },
      {
        created_on: '2017-10-06T16:57:17Z',
        direction: 'inbound',
        display_name: null,
        ems_ref: '4428d6bf-ab6d-4de9-a8eb-74c4aeed2595',
        enabled: null,
        end_port: 22,
        group: null,
        host_protocol: 'TCP',
        id: '463',
        name: null,
        network_protocol: 'IPV4',
        port: 22,
        required: null,
        resource_id: '82',
        resource_type: 'SecurityGroup',
        source_ip_range: '0.0.0.0/0',
        source_security_group_id: null,
        updated_on: '2017-10-06T16:57:17Z',
      },
    ];
    const editSecurityGroup = {
      cloud_tenant: {
        name: 'cloud-user-demo',
      },
      description: '',
      ems_id: '8',
      ext_management_system: {
        name: 'OpenStack Network Manager',
      },
      firewall_rules: editedFirewallRules,
      firewall_rules_delete: false,
      href: 'http://localhost:3000/api/security_groups/82',
      id: '82',
      name: 'asimonel-sg',
    };
    miqAjaxButton(
      `/security_group/update/${securityGroupId}?button=save`,
      editSecurityGroup,
      { complete: false }
    );
    expect(spyMiqAjaxButton).toHaveBeenCalledWith(
      `/security_group/update/${securityGroupId}?button=save`,
      editSecurityGroup,
      { complete: false }
    );
    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });
});
