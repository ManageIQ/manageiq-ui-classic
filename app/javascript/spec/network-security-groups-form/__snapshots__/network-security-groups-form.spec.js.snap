// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Network Router Interfaces Form Component should add security group 1`] = `
<Connect(MiqFormRenderer)
  buttonsLabels={
    Object {
      "submitLabel": "Add",
    }
  }
  canReset={false}
  canSubmit={false}
  onCancel={[Function]}
  onSubmit={[Function]}
  schema={
    Object {
      "fields": Array [
        Object {
          "component": "sub-form",
          "fields": Array [
            Object {
              "component": "select",
              "id": "ems_id",
              "isDisabled": false,
              "isRequired": true,
              "label": "Network Manager",
              "loadOptions": [Function],
              "name": "ems_id",
              "onChange": [Function],
              "placeholder": "<Choose>",
              "validate": Array [
                Object {
                  "message": "Required",
                  "type": "required",
                },
              ],
              "validateOnMount": true,
            },
          ],
          "id": "network-provider",
          "name": "network-provider",
          "title": "Network Management Provider",
        },
      ],
    }
  }
  validate={[Function]}
/>
`;

exports[`Network Router Interfaces Form Component should edit security group 1`] = `
<Loading
  active={true}
  className="export-spinner"
  description="Active loading indicator"
  small={true}
  withOverlay={false}
/>
`;

exports[`Network Router Interfaces Form Component should edit security group firewall rules 1`] = `
<Loading
  active={true}
  className="export-spinner"
  description="Active loading indicator"
  small={true}
  withOverlay={false}
/>
`;

exports[`Network Router Interfaces Form Component should render add security group form 1`] = `
<Connect(MiqFormRenderer)
  buttonsLabels={
    Object {
      "submitLabel": "Add",
    }
  }
  canReset={false}
  canSubmit={false}
  onCancel={[Function]}
  onSubmit={[Function]}
  schema={
    Object {
      "fields": Array [
        Object {
          "component": "sub-form",
          "fields": Array [
            Object {
              "component": "select",
              "id": "ems_id",
              "isDisabled": false,
              "isRequired": true,
              "label": "Network Manager",
              "loadOptions": [Function],
              "name": "ems_id",
              "onChange": [Function],
              "placeholder": "<Choose>",
              "validate": Array [
                Object {
                  "message": "Required",
                  "type": "required",
                },
              ],
              "validateOnMount": true,
            },
          ],
          "id": "network-provider",
          "name": "network-provider",
          "title": "Network Management Provider",
        },
      ],
    }
  }
  validate={[Function]}
/>
`;

exports[`Network Router Interfaces Form Component should render edit security group form 1`] = `
<Loading
  active={true}
  className="export-spinner"
  description="Active loading indicator"
  small={true}
  withOverlay={false}
/>
`;
